### ISSUE TITLE
Scoop installs 7-Zip even when 7ZIPEXTRACT_USE_EXTERNAL is true

### ISSUE THREAD

**Issue by TheRandomLabs** [ScoopInstaller/Scoop#3700]
If `Expand-7ZipArchive` is detected in the installer script, 7-Zip is installed even when `7ZIPEXTRACT_USE_EXTERNAL` is set to `true`.

**Comment by TheRandomLabs** [comment]
Changing [this] line to the following seems to fix this, but I don't know if it's the right way to do it:

```powershell
if (!(get_config 7ZIPEXTRACT_USE_EXTERNAL) -and ($script -like '*Expand-7zipArchive *' -or $script -like '*extract_7zip *')) {
```

**Comment by TheRandomLabs** [comment]
I've also noticed that [here](https://github.com/lukesampson/scoop/blob/96de9c14bb483f9278e4b0a9e22b1923ee752901/lib/install.ps1#L544), `7zip-zstd` is used even when `7ZIPEXTRACT_USE_EXTERNAL` is set to `false`. Should this behavior not be made consistent?
[Relevant](https://github.com/lukesampson/scoop/blob/eb3d42de8f4eb3a4c83a2fca6e189791e246ae73/lib/depends.ps1#L61) [lines](https://github.com/lukesampson/scoop/blob/eb3d42de8f4eb3a4c83a2fca6e189791e246ae73/lib/depends.ps1#L80)

**Comment by rashil2000** [comment]
@niheaven has the fix for this landed in `master`?

**Comment by niheaven** [comment]
Not yet, it is fixed in #3502 which needs some rebasing onto current 'develop'.

**Comment by bc-185** [comment]
What about creating a pull request for this?

**Comment by rashil2000** [comment]
Closed in #4638
